### PR TITLE
NEW: Support tapers other than dpss with additional parameters

### LINKED ISSUE #183: General tapers support taperopt
All backends which use a `taper` parameter also have a `taperopt={..}`, however the frontends don't make any use of this right now, except of course for the `dpss` multi tapering. Even though we allow in `const_def.availableTapers` a bunch which have additional parameters (e.g. `gaussian`, `kaiser`, ...), for these we get `TypeErrors` at the moment. 

It should be straightforward to allow a more general `taperopt` keyword in the respective frontends (`freqanalysis`, `connectivity`).

### PR DESCRIPTION
**Pull request by tensionhead**
- tapers like 'kaiser' need additional parameters which now can be
passed from the fronted via the new `taper_opt` parameter
- no changes in backend needed as here a `taper_opt` was already present
- resolves #183

On branch general_tapers
Changes to be committed:
	modified:   syncopy/nwanalysis/connectivity_analysis.py
	modified:   syncopy/shared/const_def.py
	modified:   syncopy/shared/input_processors.py
	modified:   syncopy/specest/freqanalysis.py

Author Guidelines
-----------------
- [x] Is the change set **< 400 lines**?
The introduced changes can't effect any of those points
- [ ] Was the code checked for memory leaks/performance bottlenecks?
- [ ] Is the code running locally **and** on the ESI cluster?
- [ ] Is the code running on all supported platforms?

Reviewer Checklist
------------------
- [ ] Are testing routines present?
- [ ] Do **parallel** loops have a set length and correct termination conditions?
- [ ] Do objects in the global package namespace perform proper parsing of their input? 
- [ ] Do code-blocks provide novel functionality, i.e., no re-factoring using builtin/external packages possible?
- [ ] Code layout
  - [ ] Is the code PEP8 compliant?
  - [ ] Does the code adhere to agreed-upon naming conventions?
  - [ ] Are keywords clearly named and easy to understand?
  - [ ] No commented-out code?
- [ ] Are all docstrings complete and accurate?

### CODE REVIEW

File: syncopy/nwanalysis/connectivity_analysis.py [hunk 1]
```diff
@@ -115,6 +115,10 @@ def connectivityanalysis(data, method="coh", keeptrials=False, output="abs",
     taper : str
         Only valid if `method` is `'coh'` or `'granger'`. Windowing function,
         one of :data:`~syncopy.specest.const_def.availableTapers`
+    taper_opt : dict or None
+        Dictionary with keys for additional taper parameters.
+        For example :func:`~scipy.signal.windows.kaiser` has
+        the additional parameter 'beta'.
```
Comment (pantaray) [inline syncopy/nwanalysis/connectivity_analysis.py:121]: The `process_taper` routine is starting to get pretty complex with lots of if/else branching. If I understand correctly when utilizing DPSS tapers any user-provided `taper_opt` is ignored and instead `dpss_opt = {'NW': NW, 'Kmax': nTaper}` is used. I think this should be emphasized in the docstring.

File: syncopy/specest/freqanalysis.py [hunk 2]
```diff
@@ -165,6 +165,10 @@ def freqanalysis(data, method='mtmfft', output='pow',
     taper : str
         Only valid if `method` is `'mtmfft'` or `'mtmconvol'`. Windowing function,
         one of :data:`~syncopy.shared.const_def.availableTapers` (see below).
+    taper_opt : dict or None
+        Dictionary with keys for additional taper parameters.
+        For example :func:`~scipy.signal.windows.kaiser` has
+        the additional parameter 'beta'.
```
Comment (pantaray) [inline syncopy/specest/freqanalysis.py:171]: See comment in `connectivityanalysis`

**Review by pantaray — COMMENTED** [review]
Looks good to me - I would just clarify that `taper_opt` doesn't do anything with DPSS tapers.

File: syncopy/nwanalysis/connectivity_analysis.py (hunk 1 above)
Comment (tensionhead) [reply syncopy/nwanalysis/connectivity_analysis.py:121]: Yep, but at least it's re-usable for both `specest` and `nwanalysis` frontends ;) I am also not super happy about the current structure, but as you can see here; https://github.com/esi-neuroscience/syncopy/blob/f83188af815fe93ac84d2e18f4462cc326557c1f/syncopy/shared/input_processors.py#L246 The users are getting a warning, and `dpss` is only configurable via our `tapsmofrq` (and the not recommenden `nTaper`) 
I will amend the docstring!

Comment (pantaray) [reply syncopy/nwanalysis/connectivity_analysis.py:121]: Great, thank you! Other than that, I think we can merge right away. Thanks @tensionhead !

**Review by pantaray — APPROVED** [review]
Nice work @tensionhead Feel free to merge!

**Pull request merged**